Source/UZKArchive.h

```diff
@@ -618,6 +618,33 @@ compressionMethod:(UZKCompressionMethod)method
         overwrite:(BOOL)overwrite
             error:(NSError **)error;
 
+
+/**
+ Writes the data to the zip file, overwriting only if specified with the overwrite flag. Overwriting
+ presents a tradeoff: the whole archive needs to be copied (minus the file to be overwritten) before
+ the write begins. For a large archive, this can be slow. On the other hand, when not overwriting,
+ the size of the archive will grow each time the file is written.
+
+ @param data Data to write into the archive
+ @param filePath The full path to the target file in the archive
+ @param fileDate The timestamp of the file in the archive. Uses the current time if nil
+ @param posixPermissions the source posix permissions of the file in the archive
+ @param method The UZKCompressionMethod to use (Default, None, Fastest, Best)
+ @param password Override the password associated with the archive (not recommended)
+ @param overwrite If YES, and the file exists, delete it before writing. If NO,
+                            append the data into the archive without removing it first (legacy Objective-Zip behavior)
+ @param error Contains an NSError object when there was an error writing to the archive
+ @return YES if successful, NO on error
+ */
+- (BOOL)writeData:(NSData *)data
+         filePath:(NSString *)filePath
+         fileDate:(nullable NSDate *)fileDate
+ posixPermissions:(unsigned long)posixPermissions
+compressionMethod:(UZKCompressionMethod)method
+         password:(nullable NSString *)password
+        overwrite:(BOOL)overwrite
+            error:(NSError **)error;
```

Comment:
Hi abbeycode, i Add this method without changing the original code base until the code passes. Is this feasible?